File: src/main/java/net/arvandor/numinoustreasury/profession/NuminousProfession.java
package net.arvandor.numinoustreasury.profession;

import org.bukkit.configuration.serialization.ConfigurationSerializable;
import org.bukkit.configuration.serialization.SerializableAs;

import java.util.Map;

@SerializableAs("NuminousProfession")
public final class NuminousProfession implements ConfigurationSerializable {

    private final String id;
    private final String name;

    public NuminousProfession(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    @Override
    public Map<String, Object> serialize() {
        return Map.of(
                "id", getId(),
                "name", getName()
        );
    }

    public static NuminousProfession deserialize(Map<String, Object> serialized) {
        return new NuminousProfession(
                (String) serialized.get("id"),
                (String) serialized.get("name")
        );
    }
}

File: src/main/kotlin/net/arvandor/numinoustreasury/profession/NuminousProfession.kt
package net.arvandor.numinoustreasury.profession

import org.bukkit.configuration.serialization.ConfigurationSerializable
import org.bukkit.configuration.serialization.SerializableAs

@SerializableAs("NuminousProfession")
class NuminousProfession(val id: String, val name: String) : ConfigurationSerializable {
    override fun serialize(): Map<String, Any> {
        return mapOf(
            "id" to id,
            "name" to name,
        )
    }

    companion object {
        @JvmStatic
        fun deserialize(serialized: Map<String?, Any?>): NuminousProfession {
            return NuminousProfession(
                serialized["id"] as String,
                serialized["name"] as String,
            )
        }
    }
}
